File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    q=[]
    for i in scoville:
        heapq.heappush(q,i)
    temp=0
    while q:
        now=heapq.heappop(q)
        if now>=K:
            return temp
        if len(q)==0:
            break
        now1=heapq.heappop(q)
        heapq.heappush(q,now+now1*2)
        temp+=1
    return -1
    

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer>q=new PriorityQueue<>();
        for(int i:scoville){
            q.add(i);
        }
        
        while(!q.isEmpty()){
            int now=q.poll();
            if(now>=K)
                return answer;
            if(q.isEmpty())
                return -1;
            int now1=q.poll();
            q.add(now+now1*2);
            answer+=1;

        
        }

        return -1;
    }
}